validator.go

```diff
@@ -84,49 +87,15 @@ func (m TypeValidator) Validate(msg Message) error {
 // NewTypeValidator is a TypeValidator factory.
 func NewTypeValidator(m map[MessageType]Validator, defaultValidators ...Validator) (TypeValidator, error) {
```

Comment:
I would just accept one `defaultValidator Validator` here.  If someone wants to put in `Validators`, they can build it themselves pretty easily and pass it in.